lib/ansible/modules/cloud/amazon/aws_s3_bucket_facts.py

```diff
@@ -0,0 +1,172 @@
+        "server": "AmazonS3",
+        "transfer-encoding": "chunked",
+        "x-amz-id-2": "LeNLtFwr0DmQD7+pb52w3A+vuR9vkVNDswfJyTHj6673yBDoYS24EShCztue+ARrxqzDGsNWmOk=",
+        "x-amz-request-id": "F47C7B7622231E44"
+      },
+      "HTTPStatusCode": 200,
+      "HostId": "LeNLtFwr0DmQD7+pb52w3A+vuR9vkVNDswfJyTHj6673yBDoYS24EShCztue+ARrxqzDGsNWmOk=",
+      "RequestId": "F47C7B7622231E44",
+      "RetryAttempts": 0
+    }
+'''
+
+import json
+import traceback
+from datetime import datetime
+from json import dumps
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import get_aws_connection_info, ec2_argument_spec, boto3_conn, \
+    camel_dict_to_snake_dict
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, NoCredentialsError
+
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+class S3ServiceManager(object):
+    """
+    Service manager for S3
+    """
+
+    def __init__(self, module):
+        """
+        Bootstrap module and connection
+        :param module:
+        """
+        self.module = module
+
+        # Set up connection
+        region, ec2_url, aws_connect_params = get_aws_connection_info(self.module, boto3=HAS_BOTO3)
+
+        if region:
+            try:
+                self.connection = boto3_conn(self.module, conn_type='client', resource='s3', region=region,
+                                             endpoint=ec2_url,
+                                             **aws_connect_params)
+            except ClientError as e:
+                self.module.fail_json(msg=e.message, exception=traceback.format_exc(),
+                                      **camel_dict_to_snake_dict(e.response))
+        else:
+            self.module.fail_json(msg="AWS region must be specified (like: us-east-1)")
+
+    def get_bucket_list(self):
+        """
+        Return result of list_buckets json encoded
+        :return:
+        """
+        try:
+            buckets = self.connection.list_buckets()
+        except ClientError as e:
+            self.module.fail_json(msg=e.message, exception=traceback.format_exc(),
+                                  **camel_dict_to_snake_dict(e.response))
+
+        return dumps(buckets, cls=DateTimeEncoder)
+
+
+class DateTimeEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, datetime):
+            return o.isoformat()
+        return json.JSONEncoder.default(self, o)
+
+
+def main():
+    """
+    Run module
+    :return:
+    """
+
+    # Including ec2 argument spec, no support for check mode as it will always return new results
+    module = AnsibleModule(argument_spec=ec2_argument_spec(), supports_check_mode=False)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    # Set defaults
+    result = {'s3': {}}
+
+    result['changed'] = False
+    result['s3'] = S3ServiceManager(module).get_bucket_list()
```

Comment:
Since this is the only key in the dict that provides helpful info you can unnest it for ease of use. `result = S3ServiceManager(module).get_bucket_list()`